src/MongoDB.Driver.Core/Core/WireProtocol/Messages/Encoders/BinaryEncoders/CommandMessageBinaryEncoder.cs

```diff
@@ -313,7 +310,20 @@ private void WriteType1Section(BsonBinaryWriter writer, Type1CommandMessageSecti
                     serializer.Serialize(context, document);
 
                     var messageSize = stream.Position - messageStartPosition;
-                    if (messageSize > maxMessageSize && batch.CanBeSplit)
+                    bool shouldBeSplit;
+                    if (IsEncryptionConfigured && MaxEncryptionDocumentSize.HasValue)
+                    {
+                        shouldBeSplit =
+                            messageSize > MaxEncryptionDocumentSize &&
+                            maxBatchCount > 1 &&
+                            i != 0;
```

Comment:
It looks like the first element should always be not split. 
I'm not 100% sure about this condition. But in case if we remove this condition generated `documents` node can be generated empty in some cases.
cc: @rstam

Reply:
Empty splits are not allowed.